spec-cordova/prepare.spec.js

```diff
@@ -17,175 +17,265 @@
-                    [ 'before_prepare',
-                    { verbose: false,
-                        platforms:
-                         [ 'ios',
-                           'osx',
-                           'android',
-                           'ubuntu',
-                           'blackberry10',
-                           'windows',
-                           'webos',
-                           'browser' ],
-                        options: {},
-                        save: false,
-                        fetch: false,
-                        paths:
-                         [ path.join('/','some','path','platforms','ios','www'),
-                           path.join('/','some','path','platforms','osx','www'),
-                           path.join('/','some','path','platforms','android','www'),
-                           path.join('/','some','path','platforms','ubuntu','www'),
-                           path.join('/','some','path','platforms','blackberry10','www'),
-                           path.join('/','some','path','platforms','windows','www'),
-                           path.join('/','some','path','platforms','webos','www'),
-                           path.join('/','some','path','platforms','browser','www') ],
-                        searchpath: undefined } ]);
-                }, function(err) {
-                    expect(err).toBeUndefined();
-                }).fin(done);
+    describe('restoreMissingPluginsForPlatform helper method', function () {
+        let is_plugin_installed_mock;
+        let is_plugin_provider_get_mock;
+        it('should resolve quickly if there is no difference between "old" and "new" platform.json', function(done) {
+            is_plugin_installed_mock = jasmine.createSpy('is plugin installed mock');
+            // mock platform json value below
+            PlatformJson.load.and.callFake(function(platformJsonPath, plat) {
+                return {
+                    isPluginInstalled: is_plugin_installed_mock,
+                    root: {
+                        installed_plugins: [],
+                        dependent_plugins: []
+                    }
+                };
             });
-            it('Test 006 : should fire after hooks through the hooker module, and pass in platforms and paths as data object', function(done) {
-                prepare('android').then(function() {
-                    expect(fire.calls.argsFor(1)).toEqual([ 'after_prepare',{ platforms: [ 'android' ],verbose: false,options: {},paths: [ path.join( '/','some','path', 'platforms', 'android', 'www' ) ],searchpath: undefined } ]);
-                }, function(err) {
-                    expect(err).toBeUndefined('Exception while running `prepare android`:\n' + err.stack);
-                }).fin(done);
+
+            prepare.restoreMissingPluginsForPlatform('android', project_dir, {}).then(function() {
+                expect(platforms.getPlatformApi).not.toHaveBeenCalled();
```

Comment:
If I understand correctly, we are asserting that `getPlatformApi` does not get called as a way of ensuring that `restoreMissingPluginsForPlatform` "short-circuits" out of the heavier logic, in the easy case. If that is true, can we change the test name to reflect that so that it's clear for people looking at these in the future?

Perhaps "should resolve quickly and not invoke getPlatformAPI in the easy case of there being no different between old and new platform.json" ?